ext/standard/array.c

```diff
@@ -6349,9 +6353,24 @@ PHP_FUNCTION(array_key_exists)
 		case IS_NULL:
 			RETVAL_BOOL(zend_hash_exists_ind(ht, ZSTR_EMPTY_ALLOC()));
 			break;
+		case IS_DOUBLE:
```

Comment:
Isn't it a problem that the `IS_STRING` branch doesn't handle numeric strings separately like how zend_execute.c does?

Reply:
It uses zend_symtable, so that's okay.